core/embed/lib/fonts/fonts.c

```diff
@@ -22,127 +22,126 @@
-}
 
-int font_max_height(int font) {
-  switch (font) {
+#define PASTER(font_name) font_name##_info
+#define FONT_INFO(font_name) PASTER(font_name)
+
+
+static const font_info_t *get_font_info(font_id_t font_id) {
+  switch (font_id) {
 #ifdef TREZOR_FONT_NORMAL_ENABLE
     case FONT_NORMAL:
-      return FONT_NORMAL_MAX_HEIGHT;
-#endif
-#ifdef TREZOR_FONT_DEMIBOLD_ENABLE
-    case FONT_DEMIBOLD:
-      return FONT_DEMIBOLD_MAX_HEIGHT;
+      return &FONT_INFO(TREZOR_FONT_NORMAL_NAME);
 #endif
 #ifdef TREZOR_FONT_BOLD_ENABLE
     case FONT_BOLD:
-      return FONT_BOLD_MAX_HEIGHT;
-#endif
-#ifdef TREZOR_FONT_NORMAL_UPPER_ENABLE
-    case FONT_NORMAL_UPPER:
-      return FONT_NORMAL_UPPER_MAX_HEIGHT;
+      return &FONT_INFO(TREZOR_FONT_BOLD_NAME);
 #endif
-#ifdef TREZOR_FONT_BOLD_UPPER_ENABLE
-    case FONT_BOLD_UPPER:
-      return FONT_BOLD_UPPER_MAX_HEIGHT;
+#ifdef TREZOR_FONT_DEMIBOLD_ENABLE
+    case FONT_DEMIBOLD:
+      return &FONT_INFO(TREZOR_FONT_DEMIBOLD_NAME);
 #endif
 #ifdef TREZOR_FONT_MONO_ENABLE
     case FONT_MONO:
-      return FONT_MONO_MAX_HEIGHT;
+      return &FONT_INFO(TREZOR_FONT_MONO_NAME);
 #endif
 #ifdef TREZOR_FONT_BIG_ENABLE
     case FONT_BIG:
-      return FONT_BIG_MAX_HEIGHT;
-#endif
-#ifdef TREZOR_FONT_SUB_ENABLE
-    case FONT_SUB:
-      return FONT_SUB_MAX_HEIGHT;
-#endif
-  }
-  return 0;
-}
-
-int font_baseline(int font) {
-  switch (font) {
-#ifdef TREZOR_FONT_NORMAL_ENABLE
-    case FONT_NORMAL:
-      return FONT_NORMAL_BASELINE;
-#endif
-#ifdef TREZOR_FONT_DEMIBOLD_ENABLE
-    case FONT_DEMIBOLD:
-      return FONT_DEMIBOLD_BASELINE;
-#endif
-#ifdef TREZOR_FONT_BOLD_ENABLE
-    case FONT_BOLD:
-      return FONT_BOLD_BASELINE;
+      return &FONT_INFO(TREZOR_FONT_BIG_NAME);
 #endif
 #ifdef TREZOR_FONT_NORMAL_UPPER_ENABLE
     case FONT_NORMAL_UPPER:
-      return FONT_NORMAL_UPPER_BASELINE;
+      return &FONT_INFO(TREZOR_FONT_NORMAL_UPPER_NAME);
 #endif
 #ifdef TREZOR_FONT_BOLD_UPPER_ENABLE
     case FONT_BOLD_UPPER:
-      return FONT_BOLD_UPPER_BASELINE;
-#endif
-#ifdef TREZOR_FONT_MONO_ENABLE
-    case FONT_MONO:
-      return FONT_MONO_BASELINE;
-#endif
-#ifdef TREZOR_FONT_BIG_ENABLE
-    case FONT_BIG:
-      return FONT_BIG_BASELINE;
+      return &FONT_INFO(TREZOR_FONT_BOLD_UPPER_NAME);
 #endif
 #ifdef TREZOR_FONT_SUB_ENABLE
     case FONT_SUB:
-      return FONT_SUB_BASELINE;
+      return &FONT_INFO(TREZOR_FONT_SUB_NAME);
 #endif
+    default:
+      return NULL;
   }
-  return 0;
 }
 
-font_glyph_iter_t font_glyph_iter_init(const int font, const uint8_t *text,
+int font_height(font_id_t font_id) {
+  const font_info_t *font_info = get_font_info(font_id);
+  return font_info ? font_info->height : 0;
+}
+
+int font_max_height(font_id_t font) {
+  const font_info_t *font_info = get_font_info(font);
+  return font_info ? font_info->max_height : 0;
+}
+
+int font_baseline(font_id_t font) {
+  const font_info_t *font_info = get_font_info(font);
+  return font_info ? font_info->baseline : 0;
+}
+
```

Comment:
kind of feels like those should be removed

Reply:
all these functions are used via `ffi` from rust

Reply:
make rust call `get_font_info`? :woman_shrugging: that's the direction we want to be going

Reply:
As of this commit 9f987971882f51f7b3a3afc4675d7058b7a84dc7, the bindings to rust only contain type `font_info_t` and the function `get_font_info`.

I didn't remove them completely yet because they are used from C code.

I'll now try to remove `font_id_t` completely as you suggested.